src/components/fields/MultiTagArrayField.tsx

```diff
@@ -110,54 +139,61 @@ export default class MultiTagArrayField extends React.Component {
 		);
 	}
 
-	onChange = (key) => (formData) => {
+	onChange = memoize((key: string) => (formData: any) => {
 		const newFormData = {...this.props.formData, [key]: formData};
 		this.props.onChange(newFormData);
-	}
+	})
 
-	onButtonClick = (idx, {operation, fromField, toField}) => () => {
+	onButtonClick = memoize((idx: number) => () => {
 		const {activeButtonIdx} = this.state;
 		if (activeButtonIdx === idx) {
 			this.setState({activeButtonIdx: undefined});
 			return;
 		}
 
+		const {buttons = []} = getUiOptions(this.props.uiSchema) as Options;
+		const options = buttons[idx];
+		const {operation} = options;
+
 		if (operation === "moveAll") {
+			const {toField, fromField} = options as MoveAllButtonOptions;
 			const toFieldNewValue = (this.props.formData[toField] || []).concat(this.props.formData[fromField] || []);
 			const newFormData = {...this.props.formData, [fromField]: [], [toField]: toFieldNewValue};
 			this.props.onChange(newFormData);
 		} else if (operation === "move" || operation === "delete") {
 			this.setState({activeButtonIdx: idx});
 		}
-	}
+	})
 
-	onTagClick = (fromField) => (idx) => {
+	onTagClick = memoize((fromField: string) => (idx: number) => {
```

Comment:
You need to memoize also the fn that `(formField: string) => ` produces. Like so `memoize((fromField: string) => memoize((idx: number) => {`

Reply:
Yes that's actually correct :)